exposed _jit_pass_fold_convbn via the wrapper func fuse_conv_bn_script

edmundw314 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#39726 exposed _jit_pass_fold_convbn via the wrapper func fuse_conv_bn_script**

Differential Revision: [D21952638](https://our.internmc.facebook.com/intern/diff/D21952638)

edmundw314 committed:
exposed _jit_pass_fold_convbn via the wrapper func fuse_conv_bn_script

[ghstack-poisoned]

jerryzh168 commented on torch/quantization/quantize_script.py:
```diff
@@ -77,5 +79,5 @@ def _quantize_script(model, qconfig_dict, run_fn=None, run_args=None, inplace=Fa
 
-def quantize_dynamic_script(model, qconfig_dict, inplace=False, debug=False):
-    return _quantize_script(model, qconfig_dict, inplace=inplace, debug=debug, is_dynamic=True)
+def quantize_dynamic_script(model, qconfig_dict, sample_model_inputs, inplace=False, debug=False):
```
please revert the change

jerryzh168 commented on torch/quantization/quantize_script.py:
```diff
@@ -62,6 +63,7 @@ def _quantize_script(model, qconfig_dict, run_fn=None, run_args=None, inplace=Fa
     # copied in prepare_script when inplace is False
     if is_dynamic:
         model = prepare_dynamic_script(model, qconfig_dict, inplace)
+        model(*run_args)
```
I think you need to rebase actually

jerryzh168 replied:
do `git fetch` and then `git rebase origin/master`

jerryzh168 replied:
this is removed as well

edmundw314 committed:
Update on "exposed _jit_pass_fold_convbn via the wrapper func fuse_conv_bn_script"


Differential Revision: [D21952638](https://our.internmc.facebook.com/intern/diff/D21952638)

[ghstack-poisoned]

jerryzh168 commented on torch/quantization/quantize_script.py:
```diff
@@ -35,13 +35,15 @@ def _prepare_script(model, qconfig_dict, inplace=False, quant_type=QuantType.STA
         raise ValueError('qconfig_dict should only contain names(str) as keys.')
     scripted_qconfig_dict = script_qconfig_dict(qconfig_dict)
-    model = wrap_cpp_module(torch._C._jit_pass_fold_convbn(model._c))
+    torch._C._jit_pass_dedup_module_uses(model._c)
```
this is removed

jerryzh168 commented on torch/quantization/quantize_script.py:
```diff
@@ -35,13 +35,15 @@ def _prepare_script(model, qconfig_dict, inplace=False, quant_type=QuantType.STA
                                                                inplace,
                                                                quant_type))
-
+def fuse_conv_bn_script(model):
```
can you move this to a place before this function is used? also please keep one empty line between function definitions

jerryzh168 replied:
could you define this function in https://github.com/pytorch/pytorch/blob/master/torch/nn/utils/fusion.py?

edmundw314 committed:
Update on "exposed _jit_pass_fold_convbn via the wrapper func fuse_conv_bn_script"


Differential Revision: [D21952638](https://our.internmc.facebook.com/intern/diff/D21952638)

[ghstack-poisoned]

edmundw314 committed:
Update on "exposed _jit_pass_fold_convbn via the wrapper func fuse_conv_bn_script"


Differential Revision: [D21952638](https://our.internmc.facebook.com/intern/diff/D21952638)

[ghstack-poisoned]

jerryzh168 commented:
please don't commit submodule changes

edmundw314 committed:
Update on "exposed _jit_pass_fold_convbn via the wrapper func fuse_conv_bn_script"


Differential Revision: [D21952638](https://our.internmc.facebook.com/intern/diff/D21952638)

[ghstack-poisoned]

edmundw314 committed:
Update on "exposed _jit_pass_fold_convbn via the wrapper func fuse_conv_bn_script"


Differential Revision: [D21952638](https://our.internmc.facebook.com/intern/diff/D21952638)

[ghstack-poisoned]
